Src/library/net/abstract/socket/socket.e

```diff
@@ -600,11 +600,15 @@ feature -- Input
 			-- Read a new 8-bit natural.
 			-- Make result available in `last_natural_8'.
 		do
-			read_to_managed_pointer (socket_buffer, 0, natural_8_bytes)
+			print("before: " + read_socket_buffer.read_natural_8_be (0).out + "%N")
```

Comment:
In a project, i have a thread for reading and another for writing on a socket but it causes miss read data because the SOCKET uses the same buffer for read and write.